lumicks/pylake/conftest.py

```diff
@@ -93,3 +104,99 @@ def configure_mpl():
+    raise if the file is not found, run with the option `--strict_reference_data`).
+
+    Examples
+    --------
+    ::
+
+        @pytest.mark.parametrize("par", [1])
+        def test_freezing(reference_data, par):
+            test_data = np.array([[1, 2, 3], [1, 2, 3]])
+
+            # Writes to a file function/freezing[parametrization descriptions].npz
+            np.testing.assert_allclose(test_data, reference_data(test_data))
+
+            # Writes to a file function/test_data[parametrization descriptions].npz
+            np.testing.assert_allclose(test_data, reference_data(test_data, test_name="test_data"))
+
+            # Writes to a file function/filename.npz. Note that when using this form, you are
+            # responsible for assembling the parameterization into the `file_name`!
+            np.testing.assert_allclose(test_data, reference_data(test_data, file_name="file_name"))
+    """
+
+    def get_reference_data(reference_data, test_name=None, file_name=None):
+        """Read or write reference data
+
+        Reads or writes reference data at the location `reference_data/test_name/dataset_name`.
+        This function reads and returns reference data if it is available at the provided location.
+        If it is not available, it writes a file containing the data (unless strict is enforced
+        through `pytest` in which case it throws).
+
+        `dataset_name` depends on the input arguments. By default, `dataset_name` will evaluate to
+        the name of the test from which this function is called. The arguments `test_name` and
+        `file_name` can override this behavior (see below).
+
+        Parameters
+        ----------
+        reference_data : anything
+            Data to store.
+        test_name : str, optional
+            If this is provided, the dataset name will be a combination of this string with the
+            parametrization.
+        file_name : str, optional
+            If this is provided the dataset name will be exactly the file_name without the
+            parametrization suffixed.
+
+        Raises
+        ------
+        ValueError
+            If both test_name and file_name are provided.
+        """
+        if test_name and file_name:
+            raise ValueError("You cannot specify both a test_name and file_name")
+
+        calling_function = request.function.__name__.replace("test_", "")
+
+        # Force output file name.
+        output_fn = request.node.name.replace("test_", "") if file_name is None else file_name
+
+        # Overwrite only the identifier part Test[par1-par2-par3] -> mytest[par1-par2,par3].
+        if test_name is not None:
+            output_fn = output_fn.replace(calling_function, test_name)
+
+        calling_path_name = os.path.dirname(request.fspath)
+
+        reference_data_path = os.path.join(calling_path_name, "reference_data", calling_function)
+        reference_data_fn = os.path.join(reference_data_path, f"{output_fn}.npz")
```

Comment:
Nitpick: I was a bit confused since I'm used to `fn` being short for `function`, not `filename`. E.g. `fn` is a keyword meaning `function` in Rust, and a common API name in the C++ `std` library. Consider `ref_data_filename` instead.

Reply:
Good to know. Will do! 👍